bentoml/_internal/runner/runner.py

```diff
@@ -91,25 +96,27 @@ class BatchOptions:
 
 
 class BaseRunner:
-    EXIST_NAMES: t.Set[str] = set()
-
-    def __init__(
-        self,
-        name: t.Union[str, Tag],
-        resource_quota: t.Optional[t.Dict[str, t.Any]] = None,
-        batch_options: t.Optional[t.Dict[str, t.Any]] = None,
-    ):
-        # probe an unique name
-        if isinstance(name, Tag):
-            name = name.name
-        if not name.isidentifier():
-            name = VARNAME_RE.sub("_", name)
-        self.name = name
-        self.resource_quota = ResourceQuota(
-            **(resource_quota if resource_quota else {})
-        )
-        self.batch_options = BatchOptions(**(batch_options if batch_options else {}))
-        self._impl_provider = SingletonFactory(create_runner_impl, self)
+    """
+    This class should not be implemented directly. Instead, implement the SimpleRunner or Runner.
```

Comment:
This class will not be exposed to user right?

```suggestion
    Base class for SimpleRunner or Runner.
```